packages/wallets/src/wallets/wallet-factory.ts

```diff
@@ -6,14 +6,18 @@ import type {
     CreateWalletParams,
     GetWalletSuccessResponse,
     RegisterSignerPasskeyParams,
+    DelegatedSigner as DelegatedSignerResponse,
 } from "../api";
 import { WalletCreationError, WalletNotAvailableError } from "../utils/errors";
 import type { Chain } from "../chains/chains";
 import type { InternalSignerConfig, SignerConfigForChain } from "../signers/types";
 import { Wallet } from "./wallet";
 import { assembleSigner } from "../signers";
-import type { WalletArgsFor, WalletOptions } from "./types";
-import { compareSignerConfigs } from "@/utils/signer-validation";
+import type { DelegatedSigner, WalletArgsFor, WalletOptions } from "./types";
+import { compareSignerConfigs } from "../utils/signer-validation";
+
+const DELEGATED_SIGNER_MISMATCH_ERROR =
+    "When the 'delegatedSigners' argument is provided to a method that fetches an existing wallet, that wallet must have matching delegated signers.";
```

Comment:
Why introduce this limitation? I think it should be okay if we allow only passing the delegated signers on original creation. Especially if this is used server side where the function we support is only createWallet

Reply:
adjusted